docs/cloud/pouta/api-access.md

```diff
@@ -0,0 +1,140 @@
+          "password": "'$OS_PASSWORD'"
+        }
+      }
+    },
+    "scope": {
+      "project": {
+        "domain": {"id": "'$OS_PROJECT_DOMAIN_ID'"},
+        "name": "'$OS_PROJECT_NAME'"
+      }
+    }
+  }
+}' "$OS_AUTH_URL/auth/tokens?nocatalog" | python -m json.tool
+```
+
+And obtaining your token as an `X-Subject-Token` response header. The response 
+body for our request will also contain additional useful information which includes 
+the expiration date and time of the token as `"expires_at":"datetime"`.  
+
+Once authenticated, we can make further CRUD requests to the various APIs handling our 
+cloud resources. For example, we can request the compute API for the list of available 
+virtual machine flavors as:
+
+```
+export OS_TOKEN=<copy-your-token-here>
+export OS_COMPUTE_API=https://pouta.csc.fi:8777/v2.1
+```
+
+```
+curl -s -H "X-Auth-Token: $OS_TOKEN" \
+  $OS_COMPUTE_API/flavors \
+  | python -m json.tool
+```
+
+You can consult the [Pouta web interface](https://pouta.csc.fi/dashboard/project/api_access/) further 
+to get the right values for the Pouta API endpoints such as `OS_COMPUTE_API`.  
+
+### Pouta access through client libraries
+
+[Openstacksdk](https://docs.openstack.org/openstacksdk/latest/) is a client library (SDK) for 
+building applications and services that work with OpenStack Clouds. It provides a consistent and complete 
+set of features to interact with the various OpenStack components. The SDK implements Python 
+bindings to the OpenStack API, which enables you to perform automation tasks in Python by 
+making calls on Python objects rather than making REST calls directly.
+
+In order to use it with our applications we need to first install the SDK as: 
+
+`pip install openstacksdk`
+
+Next, we need to provide our configurations and credentials through a `clouds.yaml` file which can be downloaded from 
+the [Pouta web interface](https://pouta.csc.fi/dashboard/project/api_access/) after logging in. 
+Openstacksdk expects this file in one of the following folders: the `current` directory, the `~/.config/openstack` 
+directory, or `/etc/openstack` directory. For quick test purposes, you can add your password in the `clouds.yaml` as: 
+
+```
+clouds:
+  openstack:
+    auth:
+      auth_url: https://pouta.csc.fi:5001/v3
+      username: "username"
+      password: XXXXXXXXXXX
+      project_id: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+      project_name: "project_xxxxxx"
+      user_domain_name: "Default"
+    regions:
+    - regionOne
+    interface: "public"
+    identity_api_version: 3
+```
+
+You can also split out your secrets such as the password field into a separate optional file 
+named `secure.yaml` and placed in the same folder as your `clouds.yaml` file as: 
+
+```
+clouds:
+  openstack:
+    auth:
+      password: XXXXXXXXXX
+```
+
+Now, you can run the following simple example which lists the available virtual machine flavors:
+
+```
+import openstack
```

Comment:
Missing shebang, small detail :)

Reply:
May be some users want to use the snippet in their functions or test it our in an interpreter so we can leave that detail for the user to choose.

Reply:
The interpreter does not care:

```
$ python3           
Python 3.6.8 (default, Nov 16 2020, 16:55:22) 
[GCC 4.8.5 20150623 (Red Hat 4.8.5-44)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> #!/bin/python3
... 
>>> import openstack
>>> 
```

It is important to put the shebang to mark if python2 or 3 should be used. Now it works with both, but that may not be the case in the future.